api/v1alpha3/awsmachinetemplate_webhook.go

```diff
@@ -60,8 +61,40 @@ func (r *AWSMachineTemplate) ValidateCreate() error {
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *AWSMachineTemplate) ValidateUpdate(old runtime.Object) error {
+	newAWSMachineTemplateUnstructured, err := runtime.DefaultUnstructuredConverter.ToUnstructured(r)
+	if err != nil {
+		return apierrors.NewInvalid(GroupVersion.WithKind("AWSMachineTemplate").GroupKind(), r.Name, field.ErrorList{
+			field.InternalError(nil, errors.Wrap(err, "failed to convert new AWSMachineTemplate to unstructured object")),
+		})
+	}
+	oldAWSMachineTemplateUnstructured, err := runtime.DefaultUnstructuredConverter.ToUnstructured(old)
+	if err != nil {
+		return apierrors.NewInvalid(GroupVersion.WithKind("AWSMachineTemplate").GroupKind(), r.Name, field.ErrorList{
+			field.InternalError(nil, errors.Wrap(err, "failed to convert old AWSMachineTemplate to unstructured object")),
+		})
+	}
+
 	oldAWSMachineTemplate := old.(*AWSMachineTemplate)
-	if !reflect.DeepEqual(r.Spec, oldAWSMachineTemplate.Spec) {
+
+	newAWSMachineTemplateSpec := newAWSMachineTemplateUnstructured["spec"].(map[string]interface{})
+	oldAWSMachineTemplateSpec := oldAWSMachineTemplateUnstructured["spec"].(map[string]interface{})
+
+	newAWSMachineSpec := newAWSMachineTemplateSpec["template"].(map[string]interface{})["spec"].(map[string]interface{})
+	oldAWSMachineSpec := oldAWSMachineTemplateSpec["template"].(map[string]interface{})["spec"].(map[string]interface{})
+
+	// Allow setting of cloudInit.secureSecretsBackend to "secrets-manager" only to handle v1alpha3 upgrade
+	if oldAWSMachineTemplate.Spec.Template.Spec.CloudInit.SecureSecretsBackend == "" && r.Spec.Template.Spec.CloudInit.SecureSecretsBackend == SecretBackendSecretsManager {
```

Comment:
Why do we use unstructured here?
We can set `r.Spec.Template.Spec.CloudInit.SecureSecretsBackend = "" instead?

Reply:
Not sure. Going to blame doing it late at night...
Fixed.